### PR TITLE
Add CBMC proof of base64Decode function

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of base64Decode function

Description
-----------
Add CBMC proof files for base64Decode function

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/base64Decode/base64Decode_harness.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    size_t destLen;
+    size_t * pResultLen;
+    uint8_t * pEncodedData;
+    size_t encodedLen;
+
+    size_t resultLen;
+
+    /* the base64decode function is used to decode the PEM signature and
+     * the maximum size of the decoded signature is 256. */
+    __CPROVER_assume( destLen <= MAX_DECODE_BUFFER_SIZE );
```
[the remark this answers is not in the document]
Comment (Bujain) [reply test/cbmc/proofs/base64Decode/base64Decode_harness.c:73]: It does not break the code. The base64decode function is used to decode the PEM signature and the length of the PEM signature does not exceed 256. This limit was kept inorder to have a finite limit on the loop unwinding at [line](https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/main/source/ota_base64.c#L445)

File: test/cbmc/proofs/base64Decode/base64Decode_harness.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    /* the maximum size of the encoded buffer is calculate to fit. */
+    __CPROVER_assume( encodedLen <= MAX_ENCODE_BUFFER_SIZE );
```
[the remark this answers is not in the document]
Comment (Bujain) [reply test/cbmc/proofs/base64Decode/base64Decode_harness.c:76]: Replied on the previous comment.

File: test/cbmc/proofs/base64Decode/base64Decode_harness.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (yanjos-dev) [reply test/cbmc/proofs/base64Decode/base64Decode_harness.c:73]: Oh okay that makes sense. Would it make sense to update the comment with this context then? When I read the comment originally, I was thinking that it was saying that the function does not allow for data larger than 256.

File: test/cbmc/proofs/base64Decode/base64Decode_harness.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    size_t destLen;
+    size_t * pResultLen;
+    uint8_t * pEncodedData;
+    size_t encodedLen;
+
+    size_t resultLen;
+
+    /* the base64decode function is used to decode the PEM signature and
+     * the maximum size of the decoded signature is 256.This limit is kept
+     * to have a control on unwinding of loop in base64Decode function. */
```
Comment (yanjos-dev) [inline test/cbmc/proofs/base64Decode/base64Decode_harness.c:74]:
```suggestion
    /* The base64Decode function is used to decode the PEM signature and
     * the maximum size of the decoded signature is 256. This limit is set
     * to limit loop unwinding in the base64Decode function. */
```

File: test/cbmc/proofs/base64Decode/base64Decode_harness.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (yanjos-dev) [reply test/cbmc/proofs/base64Decode/base64Decode_harness.c:78]: Can you update this comment as well?

[the remark this answers is not in the document]
Comment (Bujain) [reply test/cbmc/proofs/base64Decode/base64Decode_harness.c:78]: Done.

File: test/cbmc/proofs/base64Decode/base64Decode_harness.c (hunk 3 above)
Comment (Bujain) [reply test/cbmc/proofs/base64Decode/base64Decode_harness.c]: Done.

**Review by yanjos-dev — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**